File: app/src/main/java/com/v2/desafionubank/ui/activity/ErrorActivity.java
package com.v2.desafionubank.ui.activity;


import android.os.Bundle;
import android.widget.TextView;

import com.v2.desafionubank.R;
import com.v2.desafionubank.ui.BaseMain;
import com.v2.desafionubank.ui.NavigationManager;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * Created by CaioSChristino on 18/04/18.
 */

public class ErrorActivity extends BaseMain {
    @BindView(R.id.about_error)
    TextView mAboutError;

    @Override
    public int layoutResID() {
        return R.layout.activity_erro;
    }

    @Override
    public void initViews() {
        Bundle b = getIntent().getExtras();
        String error = b.getString(ERROR_MESSAGE);
        mAboutError.setText(error);
    }

    @OnClick(R.id.close_btn)
    public void onClickClose() {
        onBackPressed();
    }

    @Override
    public NavigationManager.BackFragmentNavigationListener setBackNavigationListener() {
        return new NavigationManager.BackFragmentNavigationListener() {
            @Override
            public void doInFinishBackNavigation() {
                finish();
            }
        };
    }
}

File: app/src/main/java/com/v2/desafionubank/ui/activity/ErrorActivity.kt
package com.v2.desafionubank.ui.activity

import com.v2.desafionubank.R
import com.v2.desafionubank.ui.BaseMain
import com.v2.desafionubank.ui.NavigationManager
import kotlinx.android.synthetic.main.activity_erro.*

/**
 * Created by csanchez on 19/04/2018.
 */

class ErrorActivity : BaseMain() {
    override fun layoutResID(): Int {
        return R.layout.activity_erro
    }

    override fun initViews() {
        val b = intent.extras
        val error = b!!.getString(ERROR_MESSAGE)
        about_error.text = error

        close_btn.setOnClickListener{
            onBackPressed()
        }
    }

    override fun setBackNavigationListener(): NavigationManager.BackFragmentNavigationListener {
        return object : NavigationManager.BackFragmentNavigationListener {
            override fun doInFinishBackNavigation() {
                finish()
            }
        }
    }
}